Fix heritability reports

vlagrassa opened the pull request:
- Minor updates to reports page so it still displays even with missing data
- Revert to accepting any trait name as second column header (in input data)

vlagrassa committed:
Accept any trait header value for nemascan

vlagrassa committed:
Link errored reports to results page

vlagrassa committed:
Add guards to ensure result defined

vlagrassa committed:
Fix indentation

vlagrassa committed:
Bugfix: add "as" clause to except

vlagrassa committed:
Add species field to heritability jobs

vlagrassa committed:
Fix typo fnam > fname

vlagrassa committed:
Prepare for header validator functions, if necessary

r-vieira commented on src/modules/site-v2/base/views/tools/heritability_calculator.py:
```diff
@@ -239,7 +239,7 @@ def report(id):
   # If report hasn't finished computing yet, display a waiting page
-  except UnfinishedReportError:
+  except UnfinishedReportError as ex:
     data, result = ex.data, None
```
@vlagrassa is there a less convoluted way (non-exception-based) way of passing data to the frontend in this code path?

vlagrassa replied:
Fixed in e67d52b0afcffaa16bb95778160f86fb855e7fde.  Waiting on PR to figure out empty Indel Finder results.

r-vieira reviewed (approved):
Looks good -- see inline comment about populating the data from the exception.

vlagrassa committed:
Cleaner handling of empty results files

Returns as None directly, rather than using `UnfinishedReportError` class. Standardizes downloading blobs to Pandas DataFrame & JSON objects.

Removes `UnfinishedReportError` class.

vlagrassa committed:
Merge branch 'development' into enhancement/job-submission-flow

vlagrassa committed:
Use description as error message in InternalError subclasses

orzut committed:
fixed downloading data for each species in data releases

orzut committed:
Merge branch 'enhancement/job-submission-flow' into bugfix/report-list-search

orzut committed:
fixing search in reports

vlagrassa committed:
Merge pull request #203 from AndersenLab/bugfix/report-list-search

Bugfix/report list search

r-vieira reviewed (approved):
looks good!
